swarm/pss/pss.go

```diff
@@ -75,15 +75,18 @@ type PssParams struct {
 }
 
 // Sane defaults for Pss
-func NewPssParams(privatekey *ecdsa.PrivateKey) *PssParams {
+func NewPssParams() *PssParams {
 	return &PssParams{
 		MsgTTL:              defaultMsgTTL,
 		CacheTTL:            defaultDigestCacheTTL,
-		privateKey:          privatekey,
 		SymKeyCacheCapacity: defaultSymKeyCacheCapacity,
 	}
 }
 
+func (self *PssParams) Init(privatekey *ecdsa.PrivateKey) {
```

Comment:
if there's no other init code here (and you don't implement as `WithPrivateKey`) please rename to a setter `SetPrivateKey`.